internal/tasks/device_render.go

```diff
@@ -292,6 +292,47 @@ func (t *DeviceRenderLogic) renderConfigItem(ctx context.Context, configItem *ap
 	}
 }
 
+// denyUnsafeDevicePath returns an error if the given absolute device path targets
+// agent-managed or read-only directories/files that must not be written by config providers.
+// Denied paths:
+//   - /var/lib/flightctl (and all subpaths)
+//   - /usr/lib/flightctl (and all subpaths)
+//   - /etc/flightctl/certs (and all subpaths)
+//   - /etc/flightctl/config.yaml (file itself)
+//
+// The check denies the exact file and the roots listed above and any subpath under those roots.
+func denyUnsafeDevicePath(p string) error {
+	// Only single absolute Linux paths are allowed in rendered configs
+	if p == "" || !filepath.IsAbs(p) {
+		return fmt.Errorf("invalid device path (must be absolute): %q", p)
+	}
+	// Reject PATH-like lists which could alter semantics downstream
+	if strings.ContainsRune(p, ':') {
+		return fmt.Errorf("invalid device path (must not contain ':'): %q", p)
+	}
+	clean := filepath.Clean(p)
+
+	// Always-denied roots
+	deniedRoots := []string{
+		"/var/lib/flightctl",
+		"/usr/lib/flightctl",
+		"/etc/flightctl/certs",
```

Comment:
I'm just curious what the use case is for allowing writing to `/etc/flightctl/` is if we're blocking other flightctl directories.